text/0745-user-provided-cdk-init-templates.md

```diff
@@ -0,0 +1,1593 @@
+
+```
+[ ] Signed-off by API Bar Raiser @xxxxx
+```
+
+
+
+## Public FAQ
+
+### What are we launching today?
+
+A new feature in the AWS CDK CLI that enables users to use custom templates for project initialization from local files, Git repositories, or NPM packages.
+
+
+### Why should I use this feature?
+
+* CDK Developer
+    * Quickly create projects with specific components, configurations, and requirements by default in the AWS CDK CLI, which reduces setup time
+    * Static template discovery in CLI
+* CDK Template Author
+    * Write and share templates for specific use cases through Git repository or NPM package
+    * Static template discovery in CLI
+
+
+
+### Are custom templates required to be “synth-able”?
+
+No, custom templates are not required to be “synth-able”. Templates generally fall under 2 categories: libraries, which create reusable constructs, and apps, which create deployable infrastructure. Library templates do not synthesize, but are still valid custom template types. As such, the `cdk init` command does not verify that a CDK stack is defined in a custom template and does not enforce a successful `cdk synth` in order to initialize a new CDK project.
+
+
+### Can I publish dynamic custom templates to a Git repository or NPM package to be used by `cdk init`?
+
+No, dynamic templates are not directly supported or processed by the CDK CLI. However, template authors who want to have variations on templates can do the branching generation-side and use dynamic template generation tools (ie. Projen, Yeoman, Cookiecutter, etc.) to develop static templates which can be published and used by a CDK user.
+
+
+## Internal FAQ
+
+### Why are we doing this?
+
+CDK users are currently limited to initializing their project from 3 built-in templates. Custom template support within the CLI reduces project setup time by allowing CDK users to quickly create CDK projects with specific components/requirements by default.
+
+
+### Why should we *not* do this?
+
+* Increased CLI complexity because users have to learn syntax for pass in new argument types
+* CDK users/template authors can already utilize custom templates outside of the CDK ecosystem (Below options achieve same outcome)
+    * Extended cdk init command to allow passing in custom templates from source:
+        * cdk init --git-url https://github.com/codepipeline/cdk-templates —template-path ./LambdaInvoke
+    * Downloading template source and extracting files:
+        * git clone https://github.com/codepipeline/cdk-templates
+        * cp -r cdk-templates/LambdaInvoke/typescript .
+
+
+
+### What is the technical solution (design) of this feature?
+
+Extend `cdk init` with:
+
+1. Custom template source options (local/Git Repo/NPM) with all CDK supported languages
+2. Simple CLI syntax for users to run `cdk init` commands with custom templates
+3. Custom template validation to ensure a valid CDK project can be initialized
+4. Initialize project using custom template
```

Comment:
How is this different from (2)? 

Also, add a note about the proposed implementation of the static discovery mechanism. This will help the team itself evaluate the complexity and feature set it can offer.